app/routes/_public.undercut.tsx

```
@@ -1,339 +1,453 @@
+        <div className="max-w-4xl mx-auto px-4">
+          <div className="text-red-600 p-4">
+            Please select a server in the options page before configuring undercut alerts.
+          </div>
+        </div>
+      </PageWrapper>
+    )
   }
 
-  const error =
-    results && 'exception' in results ? results.exception : undefined
+  const [config, setConfig] = useState<AlertConfig>({
```

ESLint

[error] React Hook "useState" is called conditionally. React Hooks must be called in the exact same order in every component render.

(react-hooks/rules-of-hooks)